tester/src/test/scala/spinal/tester/scalatest/LatchTester.scala

```diff
@@ -4,36 +4,99 @@ import org.scalatest.funsuite.AnyFunSuite
 import spinal.core._
 import spinal.core.sim._
 
+import scala.util.Random
+
 class LatchTester extends AnyFunSuite {
   test("bad latch") {
-    SimConfig.compile(new Component {
-      val myCond   = Bool
-      val inVal    = Bits(8 bits)
-      val latchVal = Bits(8 bits)
+    var didRaise = false
 
-      when(myCond) {
-        latchVal := inVal
-      }
-    })
+    try {
+      SimConfig.compile(new Component {
+        val myCond = Bool
+        val inVal = Bits(8 bits)
+        val latchVal = Bits(8 bits)
+
+        when(myCond) {
+          latchVal := inVal
+        }
+      })
+    } catch {
+      case e: SpinalExit if e.getMessage().contains("LATCH DETECTED") =>
+        didRaise = true
+    }
+
+    assert(didRaise)
   }
 
   test("allowed latch") {
     SimConfig.compile(new Component {
-      val myCond   = Bool
-      val inVal    = Bits(8 bits)
+      definition.addComment("verilator lint_off LATCH")
```

Comment:
Having to do a manual fix is not really user-friendly…

* can it be more local to the latch (signal definition?)
* can it be added automatically with `Latch` and `LatchWhen`?

@Dolu1990 any idea?